telemetry/telemetrybuffer.go

```diff
@@ -319,14 +326,15 @@ func (tb *TelemetryBuffer) ConnectToTelemetryService(telemetryNumRetries, teleme
 	}
 }
 
-func getTelemetryServiceDirectory() (path string, dir string) {
+func getTelemetryServiceDirectory() (path string, dir string, fileExists bool) {
```

Comment:
nit: Also, it would be helpful to add a comment that describes what the function does. Since it is checking the file at couple of places, would be nice to have an  explanation.

Reply:
> Would it possible to write U.T for this function only ? I don't see any dependency on telemetry here and this would increase the code coverage as well.

I can give it bad data to ensure it fails, but I do not believe I can test happy paths without mock files in the proper directories.